src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -206,6 +210,116 @@ public CtClass apply(CtType targetType) {
         public void statement()  { }
     }
 
+    @Test
+    public void testSubstituteMethodBodyWithTemplatedInitializer() {
+        // contract: Given a block with a templated initializer, substituteMethodBody should return a
+        // new block with the initializer replaced with the value bound to the template parameter
+
+        // arrange
+        Factory factory = createFactoryWithTemplates();
+        CtClass<?> targetClass = factory.Class().create("TargetClass");
+
+        String templateExecutableName = "executable";
+        String templateVariableName = "x";
+        int initializerToSubstitute = 20;
+        CtStatement expectedStatement = factory.createLocalVariable(
+                factory.Type().integerType(), templateVariableName, factory.createLiteral(initializerToSubstitute)
+        );
+        final CtBlock<?> expectedMethodBody = factory.Code().createCtBlock(expectedStatement);
+
+        BodyWithTemplatedInitializer template = new BodyWithTemplatedInitializer();
+        template._initializer_ = initializerToSubstitute;
+
+        // act
+        CtBlock<?> substitutedMethodBody = Substitution.substituteMethodBody(
+                targetClass, template, templateExecutableName
+        );
+
+        // assert
+        assertEquals(expectedMethodBody, substitutedMethodBody);
+    }
+
+    private static class BodyWithTemplatedInitializer extends ExtensionTemplate {
+        @Parameter
+        Integer _initializer_;
+
+        public void executable() {
+            Integer x = _initializer_;
+        }
+    }
```

Comment:
Why not just reuse `StatementWithTemplatedInitializer`? Or the other way around. These two templates are so similar it seems pointless to have both.

Reply:
yeah sure